src/electionguard/group.py

```diff
@@ -17,17 +20,133 @@
 from .constants import get_large_prime, get_small_prime, get_generator
 
 
-class BaseElement(ABC, int):
+class BaseElement(ABC, Number):
     """An element limited by mod T within [0, T) where T is determined by an upper_bound function."""
 
+    value: int = 0
+
     def __new__(cls, elem: Union[int, str], check_within_bounds: bool = True):  # type: ignore
         """Instantiate ElementModT where elem is an int or its hex representation or mpz."""
         if isinstance(elem, str):
             elem = hex_to_int(elem)
         if check_within_bounds:
             if not 0 <= elem < cls.get_upper_bound():
-                raise OverflowError
-        return super(BaseElement, cls).__new__(cls, elem)
+                raise OverflowError(elem, cls.get_upper_bound())
+
+        self = super(BaseElement, cls).__new__(cls)
+        self.value = int(elem)
+
+        return self
+
+    def __getnewargs__(self) -> tuple:
+        return (BaseElement.__int__(self),)
+
+    def __int__(self) -> int:
+        return self.value
+
+    def __repr__(self) -> str:
+        return f"{int(self)}"
+
+    def __add__(self, other: Any) -> BaseElement:
+        """a + b"""
+        if isinstance(other, (self.__class__, int)):
+            return self.__class__(int(self) + int(other))
+
+        return NotImplemented
+
+    def __radd__(self, other: Any) -> BaseElement:
+        """a + b"""
```

Comment:
Should probably document this as b+a since you are adding then in the other order in the method.  This happens with other reverse functions below as well.  Minor issue.